File: 0059-spiral-matrix-ii/0059-spiral-matrix-ii.java
class Solution {
    public int[][] generateMatrix(int n) {
        int[][] ans = new int[n][n] ;

        int left = 0, top = 0, right = n-1, bottom = n-1 ;
        int x = 1 ;

        while(left <= right && top <= bottom) {
            for(int i = left ; i <= right ; i++) {
                ans[top][i] = x++ ;
            }

            top++ ;

            for(int i = top ; i <= bottom ; i++) {
                ans[i][right] = x++ ;
            }

            right-- ;

            for(int i = right ; i >= left ; i--) {
                ans[bottom][i] = x++ ;
            }

            bottom-- ;

            for(int i = bottom ; i >= top ; i--) {
                ans[i][left] = x++ ;
            }

            left++ ;
        }

        return ans ;
    }
}

File: 0059-spiral-matrix-ii/0059-spiral-matrix-ii.cpp
class Solution {
public:
    vector<vector<int>> generateMatrix(int n) {
        vector<vector<int>>ans(n, vector<int>(n, 0)) ;

        int rSt = 0, cSt = 0, rend = n-1, cend = n-1 ;

        int x = 1 ;
        while(rSt <= rend && cSt <= cend) {
            for(int i = cSt ; i <= cend ; i++) {
                ans[rSt][i] = x++ ;
            }

            rSt++ ;

            for(int i = rSt ; i <= rend ; i++) {
                ans[i][cend] = x++ ;
            }

            cend-- ;

            for(int i = cend ; i >= cSt ; i--) {
                ans[rend][i] = x++ ;
            }

            rend-- ;

            for(int i = rend ; i >= rSt ; i--) {
                ans[i][cSt] = x++ ;
            }

            cSt++ ;
        }

        return ans ;
    }
};